crashappsec/chalk
# feat(plugins): added metadata for GCP and Azure of cloud providers

## Description
This PR makes the following changes:

- Queries the GCP and Azure metadata endpoints, and updates the `_GCP_INSTANCE_METADATA` and `_AZURE_INSTANCE_METADATA` endpoints respectively.
- Exposes an `_OP_CLOUD_PROVIDER` runtime key, which currently is set to one of `aws`, `gcp`, `azure`.
- [Experimental] Exposes an `_OP_CLOUD_PROVIDER_SERVICE_TYPE` runtime key, which currently is only set for AWS, and gets one of the following values `aws_eks`, `aws_ecs`, `aws_ec2`.
- [Internal] Renames the imds plugin to `cloudMetadata` plugin to reflect its cross-cloud functionality
- [Internal] Renames the `aws` and `ec2` singletons to `cloud_provider` and `cloud_instance_hw_identifiers` respectively.

## Testing

- Download a chalk binary with the above functionality in a node running in the cloud, and run a `chalk insert` command. 
- Run `tail -1 ~/.local/chalk/chalk.log | jq` and observe that the metadata keys have been populated appropriately
